packages/astro/src/runtime/server/render/page.ts

```diff
@@ -67,6 +67,22 @@ async function bufferHeadContent(result: SSRResult) {
 	}
 }
 
+// Recursively calls component instances that might have slots to be propagated up.
+async function bufferSlottedContent(result: SSRResult, outlet: string) {
+	const iterator = result.outletPropagators.entries();
+	while (true) {
+		const { value: [name, slot] = [], done } = iterator.next();
+		if (done) {
+			break;
+		}
+		const returnValue = await slot(result);
+		// TODO: exhaust all expressions
```

Comment:
Unfortunately I've been wrestling with this logic for a long time and struggled to find a clean way to extract the `Outlet` content. This is currently broken.